docs/console/incremental-guide.md

```diff
@@ -0,0 +1,586 @@
+# Writing Console Apps: A Step by Step Guide
+
+This guide will show you how to use the .NET CLI tooling to build cross-platform console apps incrementally.  It will start with the most basic console app and eventually span multiple projects, including testing.
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
```

Comment:
@richlander misleading in what way? the "paradigm" one maybe could be differently titled, it is supposed to be a best practices kind of guide as well as some instructions on what types of apps you can write and how. 

And, to do something that @cartermp dislikes and start a philosophical discussion: a paradigm can be considered to exist wherever a certain established practice exists. There is a "paradigm" of console applications just by the virtue of being able to write console applications; the fact that they are maybe simple does in no way exclude the paradigm's existence. :)